tiatoolbox/tools/stainnorm.py

```diff
@@ -136,20 +163,70 @@ def __init__(self):
         self.extractor = RuifrokExtractor()
 
 
+class MacenkoNormaliser(StainNormaliser):
+    """Macenko stain normaliser, adapted from:
+
+    Macenko, Marc, et al. "A method for normalizing histology
+    slides for quantitative analysis." 2009 IEEE International
+    Symposium on Biomedical Imaging: From Nano to Macro. IEEE, 2009.
+
+    This class contains code inspired by StainTools
+    [https://github.com/Peter554/StainTools] written by Peter Byfield.
+
+    Examples:
+        >>> from tiatoolbox.tools.stainnorm import MacenkoNormaliser()
+        >>> norm = MacenkoNormaliser()
+        >>> norm.fit(target_img)
+        >>> norm_img = norm.transform(source_img)
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.extractor = MacenkoExtractor()
+
+
+class VahadaneNormaliser(StainNormaliser):
+    """Vahadane stain normaliser, adapted from:
+
+    Vahadane, Abhishek, et al. "Structure-preserving color normalization
+    and sparse stain separation for histological images."
+    IEEE transactions on medical imaging 35.8 (2016): 1962-1971.
+
+    This class contains code inspired by StainTools
+    [https://github.com/Peter554/StainTools] written by Peter Byfield.
+
+    Examples:
+        >>> from tiatoolbox.tools.stainnorm import VahadaneNormaliser()
+        >>> norm = VahadaneNormaliser()
+        >>> norm.fit(target_img)
+        >>> norm_img = norm.transform(source_img)
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.extractor = VahadaneExtractor()
+
+
 class ReinhardColourNormaliser:
     """Normalize a patch colour to the target image using the method of:
```

Comment:
First line ends with a period.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/45/commits/4feedc5bfe03c1224b12c1c2faebc22eb7032541